src/hitchhiker/tree/bootstrap/konserve.cljc

```diff
@@ -53,6 +53,16 @@
   [key]
   (ha/promise-chan key))
 
+(defn create-id
+  "Generate a storage ID from a content UUID.
+
+  Adds a (hexadecimal) timestamp in milliseconds to the start of the
+  key; this is so the GC can properly delete keys that don't exist in
+  the tree anymore, but also skips keys that were added after the GC
+  process begins."
+  [uuid]
+  (format "%016x.%s" (System/currentTimeMillis) uuid))
```

Comment:
I would prefer not to use custom string serialization, but plain Clojure datastructures, i.e. `[(System/currentTimeMillis) uuid]`. For many backends konserve internally uses hasch to coerce the key again to a uuid anyway and for the ones where we still store the serialized key it would not be longer in serialized form than this string serialization.

Reply:
Maybe actually using an instance data type (java.util.Date) would be better than an implicit long for milliseconds.

Reply:
That sounds totally fine. The original intent with using strings was to have IDs that could sort properly, but I did away with that in the konserve patch anyway, so just a vector of `[instant guid]` works fine too -- I'll make that change.

Reply:
Also given the discussion in datahike over forking databases, the ID could be some kind of vector clock (say, an interval tree clock) instead of a simple timestamp. The stamping mechanism can even be abstracted into a protocol, so different mechanisms could be plugged in.

Reply:
Yes, that is why I had mentioned the lattices. Interval Tree Clocks are a good idea. I know the original paper, https://haslab.uminho.pt/cbm/files/itc.pdf. What references have you taken a look at?